ui/checkboxradio.js

```diff
@@ -0,0 +1,276 @@
+					}
+				}
+			}
+		}
+	},
+
+	_enhance: function() {
+		var checked = this.element[ 0 ].checked;
+
+		this._setOption( "disabled", this.options.disabled );
+		this._updateIcon( checked );
+		this._addClass( "ui-checkboxradio", "ui-helper-hidden-accessible " );
+		this._addClass( this.label, "ui-checkboxradio-label", "ui-button ui-widget" );
+
+		if ( this.type === "radio" ) {
+			this._addClass( this.label, "ui-checkboxradio-radio-label" );
+		}
+		if ( checked ) {
+			this._addClass( this.label, "ui-checkboxradio-checked", "ui-state-active" );
+		}
+		if ( this.options.label && this.options.label !== this.originalLabel ) {
+			this.label.html( this.icon ? this.icon : "" ).append( this.options.label );
+		} else if ( this.originalLabel ) {
+			this.options.label = this.originalLabel;
+		}
+	},
+
+	widget: function() {
+		return this.label;
+	},
+
+	_getRadioGroup: function( radio ) {
+		var name = this.element[0].name,
+			form = this.element[0].form,
+			radios = $( [] );
+		if ( name ) {
+			name = name.replace( /'/g, "\\'" );
+			if ( form ) {
+				radios = $( form ).find( "[name='" + name + "']" );
+			} else {
+				radios = $( "[name='" + name + "']", this.document )
+				.filter(function() {
+					return !this.form;
+				});
+			}
+		}
+		return radios.not( this.element );
+	},
+
+	_toggleClasses: function() {
+		var checked = this.element[ 0 ].checked;
+		this._toggleClass( this.label, "ui-checkboxradio-checked", "ui-state-active", checked );
+		if ( this.options.icon && this.type === "checkbox" ) {
+			this._toggleClass( this.icon, null, "ui-icon-check", checked )
+				._toggleClass( this.icon, null, "ui-icon-blank", !checked );
+		}
+		if ( this.type === "radio" ) {
+			this._getRadioGroup()
+				.each( function(){
+					var instance = $( this ).checkboxradio( "instance" );
+
+					if ( instance ) {
+						instance._removeClass( instance.label,
+							"ui-checkboxradio-checked", "ui-state-active" );
+					}
+				});
+		}
+	},
+
+	_destroy: function() {
+		if ( this.icon ) {
+			this.icon.remove();
+		}
+	},
+
+	_setOption: function( key, value ) {
+		var original;
+		if ( key === "label" && value === null ) {
+			original = this.options[ key ];
+		}
+		this._super( key, value );
+		if ( key === "disabled" ) {
+			this._toggleClass( this.label, null, "ui-state-disabled", value );
+			this.element[ 0 ].disabled =  value;
+			return;
+		}
+		if ( key === "label" && value === null ) {
+			this.options[ key ] = original;
+		}
+		this.refresh();
+	},
+
+	_updateIcon: function( checked ) {
+		var toAdd = "ui-icon ui-icon-background ";
+
+		if ( this.options.icon ) {
+			this._addClass( this.label, null, "ui-icon-beginning" );
+			if ( !this.icon ) {
+				this.icon = $( "<span>" );
+			}
+
+			if ( this.type === "checkbox" ) {
+				toAdd += checked ? "ui-icon-check" : "ui-icon-blank";
+			} else {
+				toAdd += "ui-icon-blank";
+			}
+			this._addClass( this.icon, "ui-checkboxradio-icon", toAdd );
```

Comment:
Based on my earlier testing, this must also remove "ui-icon-check" if `checked` is false.